music21/key.py

```diff
@@ -68,13 +68,26 @@ def convertKeyStringToMusic21KeyString(textString):
     'b#'
     >>> key.convertKeyStringToMusic21KeyString('c')
     'c'
+    >>> key.convertKeyStringToMusic21KeyString('Bbb')
+    'B--'
+    >>> key.convertKeyStringToMusic21KeyString('bbb')
+    'b--'
     '''
     if textString == 'bb':
         textString = 'b-'
     elif textString == 'Bb':
         textString = 'B-'
-    elif textString.endswith('b') and not textString.startswith('b'):
-        textString = textString.rstrip('b') + '-'
+    elif len(textString) == 2 and textString[-1] == 'b':
+        textString = textString[:1] + '-'
+    elif textString.endswith('b'):
```

Comment:
why not an additional check for `elif textString in ('b', 'B'): pass` so that the keys of 'b' or 'B' don't need to go through the regex?

but I wonder if the regex is overkill.  Can't it be:

```
elif textString.endswith('b'):
     textString = textString[0] + '-' * (len(textString) - 1)
```

and assume all characters in between are 'b's.  Or if we don't want to make the assumption and let things like "E-b" or "C#b" fail, something like:

```
elif set(textString[1:]) == {'b'}:
     textString = textString[0] + '-' * (len(textString) - 1)
```